### PR TITLE
Remove deleted column marker in append mode

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: clients/shared/append.go [hunk 1]
```diff
@@ -21,10 +21,12 @@ func Append(dwh destination.DataWarehouse, tableData *optimization.TableData, op
...
 
+	cols, _ := columns.RemoveDeleteColumnMarker(tableConfig.Columns().GetColumns())
+
 	// We don't care about srcKeysMissing because we don't drop columns when we append.
 	_, targetKeysMissing := columns.Diff(
-		tableData.ReadOnlyInMemoryCols(),
-		tableConfig.Columns(),
+		tableData.ReadOnlyInMemoryCols().GetColumns(),
+		cols,
 		tableData.TopicConfig().SoftDelete,
```
Comment (Tang8330) [inline clients/shared/append.go:30]: Can we hardcode `softDelete` here to be false? It should never be a soft delete for append

**Review by Tang8330 — APPROVED** [review]

File: clients/shared/append.go [hunk 2]
```diff
@@ -21,10 +21,12 @@ func Append(dwh destination.DataWarehouse, tableData *optimization.TableData, op
 		return fmt.Errorf("failed to get table config: %w", err)
 	}
 
+	cols, _ := columns.RemoveDeleteColumnMarker(tableConfig.Columns().GetColumns())
+
 	// We don't care about srcKeysMissing because we don't drop columns when we append.
 	_, targetKeysMissing := columns.Diff(
-		tableData.ReadOnlyInMemoryCols(),
-		tableConfig.Columns(),
+		tableData.ReadOnlyInMemoryCols().GetColumns(),
```
Comment (Tang8330) [inline clients/shared/append.go:28]: Actually, shouldn't we exclude it from the source and not target?

Isn't this going to add a column `__artie_deleted` in the destination?

File: clients/shared/append.go (hunk 1 above)
Comment (nathan-artie) [reply clients/shared/append.go:30]: Yep!

File: clients/shared/append.go (hunk 2 above)
Comment (nathan-artie) [reply clients/shared/append.go:29]: Ooh yeah good catch!

**Pull request merged**